### PR TITLE
Catch exception during deployment

### PR DESCRIPTION
**Pull request by elharo**
#1977 @briandealwis

### CODE REVIEW

**Comment by elharo** [comment]
Maybe not. The dialog still shows up.

**Comment by elharo** [comment]
still trying to figure out exactly what code is sending this:


!ENTRY com.google.cloud.tools.eclipse.appengine.deploy 4 1 2017-08-04 10:17:30.599
!MESSAGE ERROR: (gcloud.app.deploy) Error Response: [13] Deployment Manager operation failed, name:

**Comment by elharo** [comment]
I wonder if this is what happens when any WorkspaceJob returns a non-OK IStatus?

**Comment by chanseokoh** [comment]
Yeah, I see any job returning WARNING or ERROR logs it. However, WARNING does not bring up a dialog. INFO and CANCEL are not logged.

**Comment by elharo** [comment]
The Dialog pops up in Worker at this line:

```
					pool.endJob(currentJob, result);
```

**Comment by briandealwis** [comment]
> I wonder if this is what happens when any WorkspaceJob returns a non-OK IStatus?

Yeah.  There are some IProgressConstants to tailor the behaviour of jobs, but nothing that seems to suppress this error unless you mark it as a system job.

**Comment by elharo** [comment]
There's a org.eclipse.ui.internal.progress.ProgressManager$ listener that does this.

**Comment by elharo** [comment]
I think what we'll need to do is return Status.OK and handle messaging the error to the user ourselves in a friendlier way.

**Comment by briandealwis** [comment]
An alternative might be to run these Jobs as part of a custom `JobGroup`.  It looks like the `ProgressManager` ignores the status of individual jobs that belong to a job group and instead uses the `JobGroup`'s result, and we can override how the `JobGroup`'s result is computed across its jobs (`computeGroupResult()`).

**Comment by elharo** [comment]
The .project changes are random so far as I can tell, and a constant annoyance, I agree. I wonder if it would be simpler just not to store them in the repo at all.

**Comment by briandealwis** [comment]
We need them for any custom builders, etc, as well as the .classpath change.  After committing these changes, we shouldn't see them again, unless we add new bundles, etc.

File: third_party/com.google.cloud.tools.eclipse.jdt.launching/.classpath [hunk 1]
```diff
@@ -2,7 +2,6 @@
 <classpath>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
-	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="src" path="tests"/>
```
[the remark this answers is not in the document]
Comment (elharo) [reply third_party/com.google.cloud.tools.eclipse.jdt.launching/.classpath]: Interesting. FWIW this is not a deliberate change, just something Eclipse chose to due during one of its innumerable reload target platform/Option-F5 cycles to try to debork the worskpace. :-(

[the remark this answers is not in the document]
Comment (briandealwis) [reply third_party/com.google.cloud.tools.eclipse.jdt.launching/.classpath]: The m2e-tycho configurator doesn't have any knowledge of other source roots and so blithely wipes out and regenerates the classpath.

**Pull request merged**